lib/grpc_web/server/rack_app.rb

```diff
@@ -22,7 +22,18 @@
 # TODO: in-mem cache the result of lazy initialization
 class GRPCWeb::RackApp < ::Rack::Builder
   def handle(service_or_class, &lazy_init_block)
-    service_class = service_or_class.is_a?(Class) ? service_or_class : service_or_class.class
+    if service_or_class.is_a?(Class)
+      service_class = service_or_class
+    elsif service_or_class.is_a?(Module)
+      if lazy_init_block.nil?
+        raise(ArgumentError, "#{service_or_class} is an abstract interface. You must also provide an initializer block.") # rubocop:disable Layout/LineLength
```

Comment:
Can we move this validation into the `ServiceClassValidator` to consolidate input validation in one place?

Reply:
I'm thinking you could change the signature to something like this:
```
  def self.validate(class_or_module, implementation_provided?:)
	if class_or_module.is_a? Module && !implementation_provided
      raise(ArgumentError, "#{class_or_module} is an abstract interface. You must also provide an initializer block.")
    unless class_or_module.include?(::GRPC::GenericService)
      raise(ArgumentError, "#{class_or_module} must 'include GenericService'")
    end
    if class_or_module.rpc_descs.size.zero?
      raise(ArgumentError, "#{class_or_module} should specify some rpc descriptions")
    end

    true
  end
```
It might even make more sense to rename it to `HandleInputValidator`

Reply:
And then in this `handle` method, you would just change the existing code to:
```
service_class = service_or_class.is_a?(Module) ? service_or_class : service_or_class.class
```

Reply:
Yes, we started with that. But I think we'd need to "evaluate" the block (if a lazy init block is provided, it'd result in loading the class, etc as an autoload) to get the value for `implementation_provided`? At which point it would somewhat defeat the purpose for this change?

Agreed, the change/if/else here looks a bit clunky.